app/models/target.rb

```diff
@@ -26,11 +26,16 @@
 #  fk_rails_...  (user_id => users.id)
 #
 class Target < ApplicationRecord
-  belongs_to :user
+  belongs_to :user, counter_cache: true
   belongs_to :topic
 
   validates :title, presence: true, uniqueness: { scope: :user }
   validates :radius, presence: true
   validates :latitude, presence: true
   validates :longitude, presence: true
+  validate :validate_targets, on: :create
```

Comment:
Since this validation will run only during creation (and also I'd prefer to avoid adding bussiness logic to the model), wdyta creating a service object to handle the creation logic?

Reply:
Nice! I created services object for creating and validating targets! @JulianPasquale